src/modules.mjs

```diff
@@ -269,22 +282,22 @@ export const LibraryManager = {
         this.library = userLibraryProxy;
       }
       pushCurrentFile(filename);
+      let preprocessedName = filename.replace(/\.\w+$/, '.preprocessed$&')
+      if (VERBOSE) {
+        preprocessedName = path.join(getTempDir(), path.basename(filename));
+      }
       try {
         processed = processMacros(preprocess(filename), filename);
-        runInMacroContext(processed, {filename: filename.replace(/\.\w+$/, '.preprocessed$&')});
+        runInMacroContext(processed, {filename: preprocessedName})
       } catch (e) {
-        error(`failure to execute js library "${filename}":`);
-        if (VERBOSE) {
-          const orig = readFile(filename);
-          if (processed) {
-            error(
-              `preprocessed source (you can run a js engine on this to get a clearer error message sometimes):\n=============\n${processed}\n=============`,
-            );
+        error(`failure to execute JS library "${filename}":`);
+        if (processed) {
+          if (VERBOSE) {
+            fs.writeFileSync(preprocessedName, processed);
+            error(`preprocessed JS saved to ${preprocessedName}`)
           } else {
-            error(`original source:\n=============\n${orig}\n=============`);
+            error('use -sVERBOSE to save preprocessed JS');
```

Comment:
> Yes, the plan is for EMCC_DEBUG=2 to print the full output to stderr, and EMCC_DEBUG=1 to just show the error and preserve the tmp files.

Would this then be something like

```py
elif os.getenv('EMCC_DEBUG') == '2':
  error(`original source:\n=============\n${orig}\n=============`);
else
  error('use -sVERBOSE to save preprocessed JS');
```
Now it looks like there is no stderr output from EMCC_DEBUG=2 mode?

Or is that something you had planned for a later PR?

Reply:
> I was planning on doing that stuff as part of followup PR as part of #25092 but if you prefer I can do it all as part of this PR?

No that's fine separately, just wanted to make sure that it's on your radar (since this PR will then temporarily take away the print-preprocessed-to-stderr support)